cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/OperationsDashboardHttpHandler.java

```diff
@@ -0,0 +1,133 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.handlers;
+
+import co.cask.cdap.api.schedule.TriggeringScheduleInfo;
+import co.cask.cdap.app.store.Store;
+import co.cask.cdap.common.BadRequestException;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.namespace.NamespaceAdmin;
+import co.cask.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import co.cask.cdap.internal.app.runtime.ProgramOptionConstants;
+import co.cask.cdap.internal.app.store.RunRecordMeta;
+import co.cask.cdap.proto.NamespaceMeta;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.ProgramRunId;
+import co.cask.cdap.proto.ops.DashboardProgramRunRecord;
+import co.cask.http.HttpResponder;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+/**
+ * {@link co.cask.http.HttpHandler} to handle program run operation dashboard and reports for v3 REST APIs
+ */
+@Singleton
+@Path(Constants.Gateway.API_VERSION_3)
+public class OperationsDashboardHttpHandler extends AbstractAppFabricHttpHandler {
+  private static final Logger LOG = LoggerFactory.getLogger(OperationsDashboardHttpHandler.class);
+  private static final Gson GSON = new Gson();
+  private final Store store;
+  private final NamespaceAdmin namespaceAdmin;
+
+  @Inject
+  public OperationsDashboardHttpHandler(Store store, NamespaceAdmin namespaceAdmin) {
+    this.store = store;
+    this.namespaceAdmin = namespaceAdmin;
+  }
+
+  // TODO: [CDAP-13351] Need to support getting scheduled program runs if start + duration is ahead of the current time
+  @GET
+  @Path("/dashboard")
+  public void readDashboardDetail(FullHttpRequest request, HttpResponder responder,
+                                  @QueryParam("start") long startTimeSecs,
+                                  @QueryParam("duration") int durationTimeSecs,
+                                  @QueryParam("namespace") Set<String> namespaces)
+    throws IOException, BadRequestException {
+    if (startTimeSecs < 0) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "'start' time cannot be smaller than 0.");
+      return;
+    }
+    if (durationTimeSecs < 0) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "'duration' cannot be smaller than 0.");
+      return;
+    }
+    Set<String> existingNamespaces;
+    try {
+      existingNamespaces = namespaceAdmin.list().stream().map(NamespaceMeta::getName).collect(Collectors.toSet());
```

Comment:
So you list all namespace? Doesn't seem like a good choice. Please open a JIRA to enhance the NamespaceQueryAdmin to allow taking a collection of namespaces.